src/NServiceBus.Core/Sagas/SagaMetaModel.cs

```diff
@@ -10,20 +11,30 @@ namespace NServiceBus.Saga
     public class SagaMetaModel : IEnumerable<SagaMetadata>
     {
         /// <summary>
-        /// Populates the model with saga metadata.
+        /// Populates the model with saga metadata from the provided collection of types.
         /// </summary>
-        /// <param name="foundSagas">Collection of Sagas metadata found.</param>
-        public void Initialize(IEnumerable<SagaMetadata> foundSagas)
+        /// <param name="availableTypes">A collection of types to scan for sagas.</param>
+        public void Initialize(IEnumerable<Type> availableTypes)
         {
-            var sagas = foundSagas.ToList();
+            Initialize(availableTypes, new Conventions());
+        }
 
-            foreach (var saga in sagas)
-            {
-                byEntityName[saga.EntityName] = saga;
-            }
+        /// <summary>
+        /// Populates the model with saga metadata from the provided collection of types.
+        /// </summary>
+        /// <param name="availableTypes">A collection of types to scan for sagas.</param>
+        /// <param name="conventions">Custom conventions to be used while scanning types.</param>
+        public void Initialize(IEnumerable<Type> availableTypes, Conventions conventions)
```

Comment:
Are we accepting breaking changes here? If so, this is not really about initializing because it can safely be called multiple times. It is more of `AddRange` type of method but not sure what should be the ideal name.

I'd suggest break this up into three parts: a repository (this class), a factory (`SagaMetadata.Create`) and an entity (`SagaMetadata`).